### PR TITLE
Allow regex in blacklist fixes #167 #542

### LINKED ISSUE #167: Allow regexes in blacklist
Twitch live streams are usually found at `https://www.twitch.tv/foo`, while VODs (replays of previously broadcast streams) are usually found at `https://www.twitch.tv/foo/v/id`. While it is common for me to want to speed up VODs, I never intentionally speed up or speed down live streams, yet it happens frequently when I accidentally hit a shortcut key.

If there is no way to detect whether a video is a live stream (#152) and create an option to disable speed controls if so, then I’d like to have the ability to blacklist the first URL type and whitelist the second one. Would it be possible for the blacklist to allow either plain strings or regexes (delimited by `/…/` I suppose)?

### PR DESCRIPTION
**Pull request by ChadBailey**
Pretty straight forward, just checks if string begins with "/" then just does a raw regex comparison if so. This does not do any validation if the regex is valid. I didn't believe that was necessary but perhaps I was mistaken

### CODE REVIEW

File: inject.js [hunk 1]
```diff
@@ -292,7 +292,12 @@
         return;
       }
 
-      var regexp = new RegExp(escapeStringRegExp(match));
+      if (match.startsWith('/')) {
+        var regexp = new RegExp(match);
```
Comment (igrigorik) [inline inject.js]: If we don't validate and this throws an exception.. the whole init function will fail, right? I think we need to guard for this here, or validate at input time (harder, but better). 

Example here: https://stackoverflow.com/questions/17250815/how-to-check-if-the-input-string-is-a-valid-regular-expression

File: options.html [hunk 2]
```diff
@@ -109,7 +109,12 @@ <h3>Other</h3>
         <input id="controllerOpacity" type="text" value="">
       </div>
       <div class="row">
-        <label for="blacklist">Blacklisted sites on which extension is disabled<br/>(one per line)</label>
+        <label for="blacklist">Blacklisted sites on which extension is disabled<br/>
+            (one per line)<br/>
+            <br/>
+            <em>Regex is supported. Be sure it is in "//g" format.<br/>
+            ie: /(.+)youtube\.com(\/*)$/gi</em>
```
Comment (igrigorik) [inline options.html]: Let's add a simple validator on the input here. If we allow bad expressions, the user will have a really hard time catching this error downstream.

Comment (ChadBailey) [reply options.html]: Makes sense, thanks for the feedback. I'll see what I can do.

File: inject.js (hunk 1 above)
Comment (ChadBailey) [reply inject.js]: You're probably right, I don't really know... I'll see if I can capture the error in the event the regex is invalid and direct it to pass to the next match.

Comment (ChadBailey) [reply inject.js]: Should be fixed in commit https://github.com/igrigorik/videospeed/pull/546/commits/570c971f9a3a1d843c18b1e71b9cb410bc590f1b

File: options.html (hunk 2 above)
Comment (ChadBailey) [reply options.html]: This is going to take a little more effort than I have time for tonight as I need to create a way to visually represent a failing regex and prevent submitting with a failure. I'll try and get something here within the next couple of days.

Comment (ChadBailey) [reply options.html]: Nevermind! I didn't notice before that there was already some signaling back to the user when they click "Save", so I just hijacked that. Seems like it's working https://github.com/ChadBailey/videospeed/commit/ac8b175a9870e56c7103df6f7637aa7b043baf66

**Pull request merged**